src/test/java/test/tesng1046/TestClassSample.java

```diff
@@ -0,0 +1,52 @@
+package test.tesng1046;
+
+import org.testng.IAlterTestName;
+import org.testng.IHookCallBack;
+import org.testng.IHookable;
+import org.testng.ITestResult;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+
+import static org.testng.Assert.assertTrue;
+
+public class TestClassSample implements IHookable {
+
+    @DataProvider (name = "dp", parallel = true)
+    public Object[][] getTestData() {
+        return new Object[][] {
+            {1}, {2}, {3}, {4}, {5}
+        };
+    }
+
+    @Test (dataProvider = "dp")
+    public void testSample1(int num) {
+        assertTrue(num != 0);
+    }
+
+    @Test (dataProvider = "dp")
+    public void testSample2(int num) {
+        assertTrue(num != 0);
```

Comment:
Yes I remember your argument and I feel its a fair ask. But if you notice this class is only a dummy test class that is intended to be fed into the actual class for testing. So this class is really playing any role in the test outcome. So instead of having an empty method body (which is what I first had, but then IntelliJ cribbed about un-used variables and I didn't like using a `System.out.println` to satisfy IntelliJ, I chose to use this). Like I said, this assertion has no role to play in the actual outcome of the unit test that runs this test.